invenio_rdm_records/services/permissions.py

```diff
@@ -27,7 +29,29 @@ class RDMRecordPermissionPolicy(RecordDraftPermissionPolicy):
     """
 
     # TODO: Change all below when permissions settled
+    # Record
     can_create = [AnyUser(), SystemProcess()]
+    can_read = [AnyUser(), SystemProcess()]
+
+    # Record - files
+    # For now, can_read_files means:
+    # - can list files
+    # - can read a file metadata
+    # - can download the file
+    can_read_files = [
+        IfRestricted(
+            'record',
+            then_=[RecordOwners()],
+            else_=[
+                IfRestricted(
```

Comment:
Comment: It should be enough to check only ``files``.

Reply:
To clarify...

Do you mean that I should implement dependent validation in the AccessSchema: https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/schemas/access.py#L70 , because it's not there right now? By dependent validation, I mean something **like** if `record == "restricted"`, then there needs to be `files == "restricted"`. Otherwise, it's not enough to check `files` only. Actually there is 3 potential implementations I see:

A- have the backend do this dependent check (see note below about how frontend needs to change too then)
B- have the backend *make* the values passed be consistent. For example, if `"record": "restricted"` and `"files": "public"` is passed in, then the backend could change `"files"` to be `"files": "restricted"` so that it is consistent.  
C- have the backend do hierarchical checking (this is what my code is doing). For example, if `"record": "restricted"` and `"files": "public"` is passed in, then the backend doesn't change `"files"` but since it always checks "record" first, it doesn't matter what the other values are.

I can see advantages to each. These scenarios also apply to embargo actually and it should be made to work similarly (embargo has `until` and `reason` that **depend** on `active`). The uncertainty here has led to.... 'complexity' on the AccessComponent side. In the end, my preference is probably for A, but since it wasn't done... I didn't know the intention :sweat_smile:   

Since this dependent validation wasn't implemented, I went with this dependent permission... Also to note, because this kind of validation on the AccessSchema wasn't implemented in the backend, the frontend protection didn't set values that respect it either. We need to revisit frontend to make sure it sends consistent field values.

Which approach should we aim for @lnielsen ?